bukkit/rpk-chat-bukkit/src/main/kotlin/com/rpkit/chat/bukkit/chatchannel/RPKChatChannelImpl.kt

```diff
@@ -190,4 +195,25 @@ class RPKChatChannelImpl(
         })
     }
 
+    private fun setSenderCharacterNamePartColor(senderMinecraftProfile: RPKMinecraftProfile) {
+        val minecraftProfileId = senderMinecraftProfile.id ?: return
+        val recordExists = plugin.database.getTable(RPKChatNameColorTable::class.java)[minecraftProfileId].join() != null
```

Comment:
Minor thing but the direct database calls could probably be done by a service, and add a layer of abstraction from the data models/logic when calling it from elsewhere in the code.

Reply:
A new service has been introduced for retrieving chat name colors!